### PR TITLE
Add support for local consul agent checking

### PR DESCRIPTION
**Pull request by roberteckert**
Direct healthchecking can be expensive at scale. Using the local
agent is a best practice.

### CODE REVIEW

File: controller/consul.go [hunk 1]
```diff
@@ -97,27 +98,73 @@ func (c *ConsulMon) queryServices() ([]*App, error) {
...
+
 	var data []interface{}
 	if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
 		return false, err
 	}
+
 	for _, nodeInfo := range data {
 		n := nodeInfo.(map[string]interface{})
-		if n["Node"] == c.node {
+		if n["Node"].(string) == c.node {
```
Comment (roberteckert) [inline controller/consul.go]: It's possible this didn't work originally, so I added a type assertion to be sure.

**Comment by roberteckert** [comment]
I made some additional "cleanup" changes that I hadn't gotten around to on my initial pass. I still have not gotten around to altering the node logic, but for services and healthchecks this should be slightly better. @mayuresh82, if you can either test or tell me how to test this, it would help out. Thanks.

**Comment by roberteckert** [comment]
Changed the return signature of `getResp` since the original `Get` call returns a pointer as well.

**Comment by mayuresh82** [comment]
I made some changes to fix some syntactical and json parsing issues. Also tested this on a local lutil host and the changes look good ! Thanks for sending this out.

**Comment by mayuresh82** [comment]
FYI testing for this is a bit involved.  Until we can get some proper integration tests, the only way to test is to start this up on a consul enabled host which also has an application registered to enable_gocast.

**Pull request merged**